cmd/ssh.go

```diff
@@ -441,75 +446,69 @@ func (cmd *SSHCmd) startServices(
 	}
 }
 
-func (cmd *SSHCmd) startProxyServices(
+func (cmd *SSHCmd) startRunnerServices(
 	ctx context.Context,
 	devPodConfig *config.Config,
 	containerClient *ssh.Client,
 	log log.Logger,
-) {
-	gitCredentials := devPodConfig.ContextOption(config.ContextOptionSSHInjectGitCredentials) == "true"
-	if !gitCredentials || cmd.User == "" || cmd.GitUsername == "" || cmd.GitToken == "" {
-		return
-	}
+) error {
+	allowGitCredentials := devPodConfig.ContextOption(config.ContextOptionSSHInjectGitCredentials) == "true"
+	allowDockerCredentials := devPodConfig.ContextOption(config.ContextOptionSSHInjectDockerCredentials) == "true"
 
 	stdoutReader, stdoutWriter, err := os.Pipe()
 	if err != nil {
-		log.Debugf("Error creating stdout pipe: %v", err)
-		return
+		return fmt.Errorf("create stdout pipe: %w", err)
 	}
 	defer stdoutWriter.Close()
 
 	stdinReader, stdinWriter, err := os.Pipe()
 	if err != nil {
-		log.Debugf("Error creating stdin pipe: %v", err)
-		return
+		return fmt.Errorf("create stdin pipe: %w", err)
 	}
 	defer stdinWriter.Close()
 
 	cancelCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	errChan := make(chan error, 1)
+	errChan := make(chan error, 2)
+	// start credentials server in workspace
 	go func() {
```

Comment:
tiny: what do you think about extracting this and following routines code to separate functions? I think it would make this a bit more readable

Reply:
fair enough :)

Reply:
tried to clean it up a bit, lmk what you think 😅